pkg/controller/kieapp/logs/logs.go

```diff
@@ -0,0 +1,102 @@
+package logs
+
+import (
+	"io"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type Logger struct {
+	Logger        logr.Logger
+	SugaredLogger *zap.SugaredLogger
+}
+
+func GetLogger(name string) *zap.SugaredLogger {
+	// Set log level... override default w/ command-line variable if set.
+	debugBool := GetBoolEnv("DEBUG") // info, debug
+
+	// The logger instantiated here can be changed to any logger
+	// implementing the logr.Logger interface. This logger will
+	// be propagated through the whole operator, generating
+	// uniform and structured logs.
+	var logger Logger
+	if debugBool {
+		logger = CreateLogger(true)
+	} else {
+		logger = CreateLogger(false)
+	}
+	logger.Logger = logf.Log.WithName(name)
+	return logger.SugaredLogger.Named(name)
+}
+
+func CreateLogger(development bool) (logger Logger) {
```

Comment:
Should these 4 functions be exported? I assumed only the first one would be, and the rest would be internal implementation and subject to change

Reply:
agree

Reply:
i've changed 2 of the function names, however the GetEnv functions still need to be exported as they are called from other packages.

Reply:
Should we move those 2 up the shared.utils or somewhere?

Reply:
that's actually where they existed previously, and it was my preference. the problem is, i was hitting import conflicts when trying to use those functions from logs. we need to use them to check for the DEBUG env var from GetLogger... and since the shared/utils packages call the logs package for logging, those functions can't exist in shared/utils. 

we could offer the functions in both locations and only export from shared location... but felt didn't want to duplicate code.